worldedit-forge/src/main/java/com/sk89q/worldedit/forge/ForgeWorld.java

```diff
@@ -427,9 +426,7 @@ private void regenForWorld(Region region, Extent extent, ServerLevel serverWorld
             BlockStateHolder<?> state = ForgeAdapter.adapt(chunk.getBlockState(pos));
             BlockEntity blockEntity = chunk.getBlockEntity(pos);
             if (blockEntity != null) {
-                net.minecraft.nbt.CompoundTag tag = new net.minecraft.nbt.CompoundTag();
-                blockEntity.save(tag);
-                state = state.toBaseBlock(NBTConverter.fromNative(tag));
+                state = state.toBaseBlock(NBTConverter.fromNative(blockEntity.saveWithId()));
```

Comment:
this method will drop ForgeData and ForgeCaps, should those be preserved? (the previous method did save them, as well as id and xyz, which matches to saveWithFullMetadata. iirc xyz isn't needed but we set the correct ones back in the output.)

Reply:
Switched to that